Update IAR project template and revert dynamic heap support

deepikabhavnani opened the pull request:
### Description

Reference template file is used from IAR8.32 installation available at path
"Embedded Workbench 8.2\arm\config\template\ConvertToIAR\template_project.ewp"
Also added trustzone configs in iar_definitions.json

Reverting dynamic heap feature till IAR 8.x is stable and supported in Mbed OS for few releases
This reverts commit 8ee29bb a645bb2 891e1f2.

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

Commit:
Add Dynamic heap support to IAR

Two new block properties available in linker configuration files:
"expanding size" and "minimum size" for IAR 8.11.2 and above

Sample:
define block HEAP with expanding size, minimum size = 16K, alignment = 8 {};

This block will expand to consume all remaining available space in
the range where it is placed. If several such blocks end up in the same
range, they will share the remaining space.
Cannot place a block with expanding size inside another block with
expanding size, inside a block with maximum size, or inside an overlay.

Commit:
Update IAR project template as per IAR 8.32

Reference template file is used from IAR8.32 installation available at path
"Embedded Workbench 8.2\arm\config\template\ConvertToIAR\template_project.ewp"

Also added trustzone configs in iar_definitions.json

ciarmcom commented:
@deepikabhavnani, thank you for your changes.
@ARMmbed/mbed-os-tools @ARMmbed/mbed-os-maintainers please review.

theotherjimmy reviewed (approved):
The size of the template diff would concern me, but it looks like the settings were just re-ordered.

cmonr commented:
This can be tested now since IAR8 should be enabled in CI (for the branch at least)

cmonr commented:
CI started

0xc0170 commented:
@deepikabhavnani Please review the failures. I assume first PR should be to disable these targets (they need updating their binaries), thus  this PR should be "preceding PR needed" ?

deepikabhavnani commented:
Preceding PR is merged to feature branch, no dependencies now. Started CI

deepikabhavnani commented:
@ARMmbed/mbed-os-test @ARMmbed/mbed-os-maintainers 

Please help me understand this issue ---
```

[Pipeline] End of Pipeline



GitHub has been notified of this commit’s build result



org.jenkinsci.plugins.workflow.cps.CpsCompilationErrorsException: startup failed:

file:/var/jenkins_home/jobs/mbed-os-ci/branches/PR-9432/builds/4/libs/mbed-os-ci/src/ciconfigurations/repositories/mbedos/FeatureARMC6BranchConfig.groovy: 13: unexpected token: toolchains @ line 13, column 45.

                                 toolchains

                                 ^
1 error

```

Commit:
Dynamic heap support will be breaking change for all targets in IAR 7.x
Reverting this feature till IAR 8.x is stable and supported in Mbed OS
for few releases

1. Revert "Add Dynamic heap support to IAR"

This reverts commit 8ee29bbfa2a4a1fd25a426a5d86391331f73e0ad.

2. Revert "Splited RAM section not compatible with dynamic HEAP."

This reverts commit a645bb2a6b0c4d82f2bf3db848f980f2fa4b1f7c.

3. Revert "Add Dynamic heap support to IAR"

This reverts commit 891e1f261b4729575f64225419acee163f23b6ae.

NirSonnenschein commented:
re-checking for latest changes: started CI

alekla01 commented:
restarted CI

deepikabhavnani commented:
@ARMmbed/mbed-os-test-team  - Please help with failure in cloud client CI

`"Couldn't allocate required resources: Allocation failed"`

deepikabhavnani commented:
@ARMmbed/mbed-os-maintainers  - Not sure what is issue with cloud client CI, since this PR is for feature branch can we merge it with rest of the checks green. 

I need to rebase feature branch for PSOC boards after this is merged.

NirSonnenschein commented:
Hi @deepikabhavnani , the error seems to be CI related  ( resource allocation timeout) : re-running the cloud CI tests.

deepikabhavnani commented:
> Hi @deepikabhavnani , the error seems to be CI related ( resource allocation timeout) : re-running the cloud CI tests.

Error still existed after re-running test, @NirSonnenschein - Please see if someone can help with this

0xc0170 commented:
I restarted it one more time and will report to CI team

0xc0170 commented:
Looks like it worked ! 💯
